File: Problem Solving/find_the difference_leet_code.java
class Solution {
    public char findTheDifference(String s, String t) {
        char result=0;
		List<Character> list1=new ArrayList<Character>();
		for (int i=0;i<s.length();i++)
		{
			list1.add(s.charAt(i));
		}
		ArrayList<Character> list_missing=new ArrayList<Character>();
		for(int i=0;i<t.length();i++){
			char letter=t.charAt(i);
			if (list1.isEmpty() || (list1.contains(letter)==false))
			{
				list_missing.add(letter);
			}
			else{if(list1.contains(letter)==true){
				int index=list1.indexOf(letter);
				list1.remove(index);
			}}
        }
        return (char)list_missing.get(0);
    }
}

File: Problem Solving/find_the_difference_leet_code.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        counter_dicti={}
        for character in s:
            if character not in counter_dicti:
                counter_dicti[character]=1
            else:
                counter_dicti[character]+=1
                
        for character in t:
            if character not in counter_dicti:
                return character
            else:
                counter_dicti[character]-=1
                if counter_dicti[character]<0:
                    return character
          
